lib/foreman/force_ssl.rb

```diff
@@ -0,0 +1,50 @@
+module Foreman
+  class ForceSsl
+    UNATTENTED_PATHS = %r{^/(unattended|userdata)/}
+
+    def self.allowed_http_actions
+      @allowed_http_actions ||= []
+    end
+
+    def self.add_allowed_http_action!(action)
```

Comment:
Shall we add deprecation warning for this?

Reply:
That's a good question. On the one hand I doubt we'll have much use for it but on the other, it's typically something that you only find out when you need it. So that would be in favor of keeping it in.

I wonder if if reduces overhead if you allow it to be nil and call `&.any?` or if that's really such a micro optimization that it's not worth it. My biggest question is the copy on write and preloading aspect. If you're not calling this method, I think it will only be instantiated on the first request. That means every worker has a different array in memory.

Also, will it keep appending on reloading in development mode or does reload there somehow clear the array?

Getting that part right (or not at all) would be in favor of dropping it.

All in all we certainly need an upgrade warning for developers who use this because the mechanism is changing.

Reply:
This is required class, so it won't reload in development. It might be "bit slow" to calculate the paths on every request, but I do not feel like reduced readability for this is worth the optimalization as in Rails times the time spend is insignificant for few paths, which is what I'd expect.

Ideally I'd want us to get into all or nothing setup, so this class would be dropped and as you've envisioned having all http communication only between proxy and client (which is better design anyway).

Reply:
`[].any?` would not be what I'd be woried about in performance :)

Reply:
After talking to @ares about this I started to wonder.

Taking a step back: previously it was controller level controlled. Now it's moving to the Rails level, so really project level. That also makes me think it should be at the engine level for plugin configs. So I'd say we may actually want something in the engine DSL **if** we do want this.

However, right now I'm leaning to not having the option at all and merging this. If during the RCs we do run into it, it'll be something we can add later. Do you agree?

Reply:
Yes I definitelly agree, let's drop this and add that properly if some plugins need that.